### ISSUE TITLE
Vercel deployment uses stale data cache ignoring previous re-validations with revalidateTag

### ISSUE THREAD

**Issue by igordanchenko** [vercel/next.js#52962]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.1.0
    Relevant Packages:
      next: 13.4.11-canary.0
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
    Next.js Config:
      output: N/A
```

### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/next-52962-w2xxc5

### To Reproduce

1) Clone https://codesandbox.io/p/sandbox/next-52962-w2xxc5 into a GitHub repository
2) Deploy GitHub repository to Vercel
3) Open the deployment URL, make note of the timestamp on the page
4) Click the Refresh button (reload the page if the data doesn't refresh on its own)
5) Make note of the timestamp on the page (it is expected to be different from the one in step 3)
6) Make any code change and commit to GitHub triggering new deployment on Vercel
7) Open the deployment URL, make note of the timestamp on the page - it's the same as in step 3 

The deployment in step 6 used stale data cache from deployment in step 2 completely ignoring data re-validation in step 4.

### Describe the Bug

Vercel deployments re-use stale data cached from previous deployments completely ignoring data re-validation performed by `revalidateTag` since previous deployment.

### Expected Behavior

New deployments must either fetch new data or use the results of the last fetch request, not the data from previous deployment which could have happened days ago.

**Comment by evankirkiles** [comment]
Found a fix, see https://github.com/vercel/next.js/issues/56370. Hopefully Vercel makes this built-in soon.

**Comment by evankirkiles** [comment]
This is still an issue. But I wrote a workaround to opt into the Vercel Data Cache here: https://github.com/evankirkiles/nextjs-broken-data-cache-demo.  Issue with explanation is here: https://github.com/vercel/next.js/issues/56370

**Comment by evankirkiles** [comment]
I have outlined the problem and a fix you can implement in your projects to opt into the Vercel Data Cache here: https://github.com/vercel/next.js/issues/56370

**Comment by ja-iy** [comment]
Hi I have experienced the same behaviour, to fetch new data on build I use

- setting the environment variable `VERCEL_FORCE_NO_BUILD_CACHE=1` ( after deploying the project )
- when redeploying with deploy hooks, `https://your-deploy-hook?buildCache=false`

This disables the build cache which also caches other things (I think). I'm guessing the reason we experience this behaviour is because the build cache and the data cache are separate.

**Comment by mspangl5** [comment]
Any updates about this issue? I'm experiencing this same issue and it is very frustrating. I don't see any other discussions happening about this topic either.

**Comment by fullyherge** [comment]
Experiencing the same issue.

My route handler:

```ts
import { revalidateTag } from "next/cache";

export const dynamic = "force-dynamic";

export async function GET() {
  revalidateTag("cms");
  return Response.json({ revalidated: true, now: Date.now() });
}

```

Works fine when running production build locally, doesn't appear to do anything in production on Vercel.

**Comment by evankirkiles** [comment]
@ztanner Is this actually closed by #58744 ? Revalidation was never the issue here—revalidation seemed to function correctly under Next.js 13—it was rather the Vercel deployment pipeline itself where previously revalidated data wouldn't show up.

If you set `NEXT_PRIVATE_DEBUG_CACHE=1` on a Vercel project, you can see that the Vercel pipeline never provides Next.js the connection details for a remote fetch cache (the Vercel Data Cache). Unless Vercel is actually hydrating the build cache's filesystem `.next/cache/fetch-cache` with updated data from the Data cache every time, I'm 95% positive that this is the issue.

With `NEXT_PRIVATE_DEBUG_CACHE=1`, we currently have in the Vercel pipeline:
```
$ next build
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry
- info Creating an optimized production build...
- info Compiled successfully
- info Linting and checking validity of types...
- info Collecting page data...
using filesystem cache handler
using filesystem cache handler
using filesystem cache handler
using filesystem cache handler
- info Generating static pages (0/22)
```

If we manually provide the connection variables for the suspense cache ourself though, we get the intended behavior and things function with the absolutely most up-to-date data, including revalidations, from the Data cache:

```
$ source ./get-sc-creds.sh; next build
Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry
   Creating an optimized production build ...
 ✓ Compiled successfully
   Linting and checking validity of types ...
   Collecting page data ...
using filesystem cache handler
using fetch cache handler
using cache endpoint https://iad1.suspense-cache.vercel-infra.com
using memory store for fetch cache
   Generating static pages (0/6) ...
```

https://github.com/evankirkiles/nextjs-broken-data-cache-demo
